service/worker/scheduler/workflow_test.go

```diff
@@ -1402,3 +1402,33 @@ func (s *workflowSuite) TestLimitedActions() {
 	s.True(s.env.IsWorkflowCompleted())
 	// doesn't end properly since it sleeps forever after pausing
 }
+
+func (s *workflowSuite) TestLotsOfIterations() {
+	// This is mostly testing getNextTime caching logic.
+	const iterations = 30
```

Comment:
one way to test my theory above is to modify this test to call backfill in between iterations 14 and 15. or maybe a series of runs that inserts a backfill at different spots (to check if it works when the cache is empty or just-refilled)

Reply:
Done